contract/src/lib.rs

```diff
@@ -424,15 +456,25 @@ impl BtcLightClient {
             .unwrap_or_else(|| env::panic_str("PrevBlockNotFound"));
 
         let current_block_hash = block_header.block_hash();
-        let pow_hash = block_header.block_hash_pow();
 
-        if !skip_pow_verification {
-            self.check_target(&block_header, &prev_block_header);
-            // Check if the block hash is less than or equal to the target
-            require!(
-                U256::from_le_bytes(&pow_hash.0) <= target_from_bits(block_header.bits),
-                format!("block should have correct pow")
-            );
+        match aux_data {
+            None => {
+                let pow_hash = block_header.block_hash_pow();
+                if !skip_pow_verification {
+                    self.check_target(&block_header, &prev_block_header);
+                    // Check if the block hash is less than or equal to the target
+                    require!(
+                        U256::from_le_bytes(&pow_hash.0) <= target_from_bits(block_header.bits),
+                        format!("block should have correct pow")
+                    );
+                }
+            }
+            Some(ref _aux_data) => {
+                #[cfg(feature = "dogecoin")]
+                self.check_aux(&block_header, _aux_data);
```

Comment:
If the aux data is optional, then what is the motivation of the relayer to submit it, can the malicious just skip it?